features/mbedtls/targets/TARGET_CRYPTOCELL310/ecdsa_alt.c

```diff
@@ -0,0 +1,325 @@
+        MBEDTLS_MPI_CHK( mbedtls_mpi_write_binary( s, pSignature + ( ( grp->nbits + 7 ) / 8 ), ( ( grp->nbits + 7 ) / 8 ) ) );
+        CrysRet =  CRYS_ECDSA_Verify ( &verifyParams->verifyContext,
+                                       &verifyParams->pubKey,
+                                       hash_mode,
+                                       pSignature,
+                                       signature_size,
+                                       (uint8_t*)buf,
+                                       blen );
+        if( CrysRet != CRYS_OK )
+        {
+            ret = convert_CrysError_to_mbedtls_err( CrysRet );
+            goto cleanup;
+        }
+    }
+    else
+        ret =  MBEDTLS_ERR_ECP_FEATURE_UNAVAILABLE;
+
+cleanup:
+
+    if( pHeap )
+    {
+        mbedtls_zeroize( pHeap, heapSize );
+        mbedtls_free( pHeap );
+    }
+
+    if( pSignature )
+    {
+        mbedtls_zeroize( pSignature, signature_size );
+        mbedtls_free( pSignature );
+
+    }
+
+    return ret;
+}
+#endif /* MBEDTLS_ECDSA_VERIFY_ALT */
+
+#if defined(MBEDTLS_ECDSA_GENKEY_ALT)
+int mbedtls_ecdsa_genkey( mbedtls_ecdsa_context *ctx, mbedtls_ecp_group_id gid,
+                  int (*f_rng)(void *, unsigned char *, size_t), void *p_rng )
+{
+    int ret = 0;
+    CRYSError_t CrysRet = CRYS_OK;
+    void* pHeap = NULL;
+    size_t heapSize = 0;
+    uint32_t key_size = 2*MAX_KEY_SIZE_IN_BYTES + 1;
+    const CRYS_ECPKI_Domain_t*  pDomain =  CRYS_ECPKI_GetEcDomain ( convert_mbedtls_grp_id_to_crys_domain_id( gid ) );
+    mbedtls_rand_func_container cc_rand = { f_rng, p_rng };
+
+
+    if ( pDomain )
+    {
+        uint8_t temp_buf[ 2 * MAX_KEY_SIZE_IN_BYTES + 1 ] = {0};
+
+        cc_ecc_ws_keygen_params_t* kgParams =  mbedtls_calloc( 1, sizeof(cc_ecc_ws_keygen_params_t) );
+        if ( kgParams == NULL )
+            return ( MBEDTLS_ERR_ECP_ALLOC_FAILED );
+
+        pHeap = kgParams;
+        heapSize = sizeof(cc_ecc_ws_keygen_params_t);
+
+        CrysRet =  CRYS_ECPKI_GenKeyPair( &cc_rand, convert_mbedtls_to_cc_rand, pDomain,
+                                          &kgParams->privKey, &kgParams->pubKey,
+                                          &kgParams->kgTempData, NULL );
+        if ( CrysRet != CRYS_OK )
+        {
+            ret = convert_CrysError_to_mbedtls_err( CrysRet );
+            goto cleanup;
+        }
+
+        MBEDTLS_MPI_CHK( mbedtls_ecp_group_load( &ctx->grp, gid ) );
+
+        CrysRet = CRYS_ECPKI_ExportPublKey( &kgParams->pubKey, CRYS_EC_PointUncompressed, temp_buf,  &key_size );
+        if ( CrysRet != CRYS_OK )
+        {
+            ret = convert_CrysError_to_mbedtls_err( CrysRet );
+            goto cleanup;
+        }
+
+        ret = mbedtls_ecp_point_read_binary( &ctx->grp, &ctx->Q, temp_buf, key_size );
+        if ( ret != 0 )
+            goto cleanup;
+
+        memset ( temp_buf, 0 , sizeof(temp_buf) );
```

Comment:
Is clearing this buffer necessary? If it is not, I would suggest removing this line, but if it is I suggest using mbedtls_zeroize as memset could be optimised away (though its very unlikely in this case.

Reply:
yes. See my answer in https://github.com/ARMmbed/mbed-os/pull/6794#discussion_r188532708
It is very unlikely in this case that it will be optimized away, if there is actual data in the buffer